contracts/GoodGhosting_Polygon.sol

```diff
@@ -71,14 +64,62 @@ contract GoodGhostingPolygon is GoodGhosting {
         // initializing incentiveController contract
         incentiveController = IncentiveController(_incentiveController);
         matic = _matic;
-        usdc = _usdc;
-        router = _router;
-        uint256 MAX_ALLOWANCE = 2**256 - 1;
-        // for the swap
-         require(
-            _matic.approve(address(_router), MAX_ALLOWANCE),
-            "Fail to approve allowance to router"
+    }
+
+    /**
+       Allowing the admin to withdraw the pool fees
+    */
+    function adminFeeWithdraw() external override  onlyOwner whenGameIsCompleted {
+        require(!adminWithdraw, "Admin has already withdrawn");
+        require(adminFeeAmount > 0, "No Fees Earned");
+        adminWithdraw = true;
+        emit AdminWithdrawal(owner(), totalGameInterest, adminFeeAmount);
+        require(
+            IERC20(daiToken).transfer(owner(), adminFeeAmount),
+            "Fail to transfer ER20 tokens to admin"
         );
+        if (rewardsPerPlayer == 0) {
+            uint balance = IERC20(matic).balanceOf(address(this));
+            require(
+                IERC20(matic).transfer(msg.sender, balance),
+                "Fail to transfer ERC20 tokens on withdraw"
+            );
+        }
```

Comment:
this logic considers that admin only get a cut from matic rewards if there's no winners in the game. it's not necessarily incorrect since we didn't discussed/defined this requirement yet.
kicked off internal thread to get the correct definition.